src/Microsoft.AspNet.Routing/Template/TemplateParser.cs

```diff
@@ -322,10 +323,27 @@ private static bool IsSegmentValid(TemplateParserContext context, TemplateSegmen
             for (var i = 0; i < segment.Parts.Count; i++)
             {
                 var part = segment.Parts[i];
+                
                 if (part.IsParameter && part.IsOptional && segment.Parts.Count > 1)
                 {
-                    context.Error = Resources.TemplateRoute_CannotHaveOptionalParameterInMultiSegment;
-                    return false;
+                    // This is the last part
+                    if (i == segment.Parts.Count - 1)
+                    {
+                        if(segment.Parts[i-1].Text == PeriodString)
+                        {
+                            part.IsFollowingAPeriod = true;
```

Comment:
Add a property `IsOptionalSeparator` to `segment.Parts[i-1]` - rather than saying `{foo} is after a period` say instead `. is optional`